### PR TITLE
[Portals] [BasicAuth] Password Mask/UnMask Toggle UX Feature

### PR DESCRIPTION
**Pull request by yasinmiran**
## Purpose
Please note $subject

## Implementation
https://user-images.githubusercontent.com/25561152/105019018-a4558280-5a6b-11eb-9a8e-b0cebbf2abff.mp4

### CODE REVIEW

**Comment by yasinmiran** [comment]
@sherenewso2 Based upon your suggestion, will this implementation suffice?

**Comment by ayshsandu** [comment]
@yasinmiran shall we place the mask/unmask toggle inside the input field like below 
image

**Comment by yasinmiran** [comment]
> @yasinmiran shall we place the mask/unmask toggle inside the input field like below
> image

+1 Working on it

**Comment by DimalChandrasiri** [comment]
@yasinmiran please check for the default ( browser injected ) password field show icon in some browsers. See if we could hide that since we have our own implementation.

File: apps/authentication-portal/src/main/webapp/basicauth.jsp [hunk 1]
```diff
@@ -421,4 +425,28 @@
...
+        /**
+         * Toggles the password visibility using the attribute
+         * type of the input.
+         *
+         * @param event {Event} click target
+         * @description stops propagation
+         */
+        $("#passwordUnmaskIcon").click(function (event) {
+            event.preventDefault();
+            const passwordInput = $("#password");
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/webapp/basicauth.jsp:440]:
```suggestion
            var passwordInput = $("#password");
```

File: apps/authentication-portal/src/main/webapp/basicauth.jsp [hunk 2]
```diff
@@ -421,4 +425,28 @@
...
+            if (passwordInput.attr("type") === "password") {
+                $(this).addClass("slash outline");
+                passwordInput.attr("type", "text");
+            } else {
+                $(this).removeClass("slash outline");
+                passwordInput.attr("type", "password");
```
Comment (jeradrutnam) [inline apps/authentication-portal/src/main/webapp/basicauth.jsp:446]:
```suggestion
            var $passwordInput = $("#password");

            if ($passwordInput.attr("type") === "password") {
                $(this).addClass("slash outline");
                $passwordInput.attr("type", "text");
            } else {
                $(this).removeClass("slash outline");
                $passwordInput.attr("type", "password");
```

**Review by jeradrutnam — APPROVED** [review]

**Pull request merged**